src/extension.js

```diff
@@ -149,271 +192,532 @@ function exportCurrentAsHtml(context, webView) {
-                inputFileGlobLimit: settings.extensionConfig().file.limit,
-                debug: settings.extensionConfig().debug,
-                repoInfo: getWsGitInfo(wsPath)
-            });
-            metrics.inputFileGlob = "{" + selectedFiles.map(x => {
-                if (x.fsPath.endsWith(".sol")) {
-                    return x.fsPath.replace(metrics.basePath, "");
-                }
-                else if (x.fsPath == wsPath || x.fsPath == metrics.basePath) {
-                    return "**/*.sol"; //special case: workspace selected
-                }
-                return x.fsPath.replace(metrics.basePath, "") + "/**/*.sol";
-            }).join(",") + "}";
-
-            vscode.window.withProgress({
-                location: vscode.ProgressLocation.Notification,
-                title: `Solidity-Metrics: crunching numbers...`,
-                cancellable: false
-            }, async (progress, token) => {
-                token.onCancellationRequested(() => {
-                    console.log("User canceled the long running operation");
-                });
+        if (!workspaceFolder) {
+          vscode.window.showErrorMessage(
+            'No workspace folder found. Please open a file or workspace.'
+          );
+          return;
+        }
 
-                //progress.report({ increment: 0 });
-                await vscode.workspace.findFiles(
-                    new vscode.RelativePattern(selectedWorkspace, metrics.inputFileGlob),
-                    new vscode.RelativePattern(selectedWorkspace, metrics.inputFileGlobExclusions),
-                    metrics.inputFileGlobLimit)
-                    .then(uris => {
-                        uris.forEach(uri => {
-                            metrics.analyze(uri.fsPath);
-                            progress.report({ increment: 1 });
-                        });
-                    });
-
-                //discover truffle projects
-                let truffleFileGlob = "{" + selectedFiles.map(x => x.fsPath.endsWith(".sol") ? x.fsPath.replace(metrics.basePath, "") : x.fsPath.replace(metrics.basePath, "") + "/**/truffle*.js").join(",") + "}";
-
-
-                await vscode.workspace.findFiles(
-                    new vscode.RelativePattern(selectedWorkspace, truffleFileGlob),
-                    new vscode.RelativePattern(selectedWorkspace, metrics.inputFileGlobExclusions),
-                    metrics.inputFileGlobLimit
-                ).then(uris => {
-                    uris.forEach(uri => {
-                        if (uri.fsPath.endsWith(".js")) {
-                            metrics.addTruffleProjectLocation(uri.fsPath);
-                            progress.report({ increment: 1 });
-                        }
-                    });
-                });
-                //list excluded files
+        // Prompt for the relative path to the scope file
+        let scopeFilePath = await vscode.window.showInputBox({
```

Comment:
what if we omit the prompt and assume that if the workspace has a "scope.txt" if you right click on it in the explorer it shows a "metrics report for scope file" menu entry or so?